core/src/main/java/org/geysermc/geyser/network/netty/GeyserServer.java

```diff
@@ -118,31 +128,33 @@ public GeyserServer(GeyserImpl geyser, int threadCount) {
     }
 
     public CompletableFuture<Void> bind(InetSocketAddress address) {
-        CompletableFuture<Void> future = new CompletableFuture<>();
-        this.bootstrapFuture = this.bootstrap.bind(address).addListener(bindResult -> {
-            if (bindResult.cause() != null) {
-                future.completeExceptionally(bindResult.cause());
-                return;
-            }
-            future.complete(null);
-        });
+        bootstrapFutures = new ChannelFuture[listenCount];
+        for (int i = 0; i < listenCount; i++) {
+            bootstrapFutures[i] = Optional.of(bootstrap)
+                    .map(f -> f.bind(address))
+                    .map(f -> addHandlers(f, this))
+                    .orElseThrow();
```

Comment:
I don't think we should use Optional here if
1. we're not going to use `ofNullable` (and it won't be anyway)
2. bind isn't going to return null
3. addHandlers just passes through the channel future